src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetRelevantDeparmentProfile.cs

```diff
@@ -0,0 +1,322 @@
+            {
+                this.logger = logger;
+                this.profileResolver = profileResolver;
+                this.rolesClient = rolesClient;
+                this.mediator = mediator;
+                this.lineOrgResolver = lineOrgResolver;
+                this.memCache = memCache;
+
+            }
+            private CancellationToken _cancellationToken;
+            public async Task<IEnumerable<QueryRelevantDepartmentProfile>> Handle(GetRelevantDeparmentProfile request, CancellationToken cancellationToken)
+            {
+                _cancellationToken = cancellationToken;
+                var user = await profileResolver.ResolvePersonBasicProfileAsync(request.ProfileId.OriginalIdentifier);
+
+                if (user is null) return null;
+
+                // Resolve departments with responsibility
+                var departmentsWithAccess = await ResolveDepartmentsWithAccessAsync(user);
+
+                if (user.IsResourceOwner)
+                {
+                    List<string> Reasons = new List<string>();
+                    Reasons.Add("Manager");
+
+                    if (user.FullDepartment != null)
+                    {
+
+                        QueryRelevantDepartmentProfile? DepartmentProfile = await GetDepartmentInformation(user.FullDepartment, Reasons, request.Query);
+                       await StoreResult(lstDepartments, DepartmentProfile);
+                    }
+                }
+
+
+                foreach (var department in departmentsWithAccess)
+                {
+                    if (sapIdFound == true)
+                    {
+                        break;
+                    }
+                    List<string> Reasons = new List<string>();
+                    if (department.Value.Contains("Resources.ResourceOwner") && !department.Key.Contains("*"))
+                    {
+
+                        Reasons.Add("DelegatedManager");
+
+                    }
+                    else if (department.Value.Contains("Resources.FullControl") && department.Key.Contains("*"))
+                    {
+
+                        List<string> reason = new List<string>();
+                        reason.Add("ParentManager");
+                        var DepChildren = await ResolveChildren(department.Key.Trim('*').TrimEnd());
+
+                        await GetDepartmentChildren(department.Key.Trim('*').TrimEnd(), reason, request.Query);
+
+
+                        Reasons.Add("Manager ( Wildcard) ");
+
+
+                    }
+                    else if (department.Value.Contains("Resources.FullControl") && !department.Key.Contains("*"))
+                    {
+                        Reasons.Add("Write Access");
+
+                    }
+                    else
+                    {
+
+                        Reasons.Add("NOT Manager");
```

Comment:
Please conform to what you have written in the SPEC-doc
NOT Manager doesn't seem like a valid option.